### ISSUE TITLE
np.equal can still return NotImplemented

### ISSUE THREAD

**Issue by shoyer** [numpy/numpy#6479]
It looks like this didn't make it into the clean up in #5864/#5964.

```
In [3]: np.equal(np.array(['a', 'b']), 123)
/Users/shoyer/miniconda/envs/numpy-dev-py3/bin/ipython:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  #!/bin/bash /Users/shoyer/miniconda/envs/numpy-dev-py3/bin/python.app
Out[3]: NotImplemented

In [4]: np.array(['a', 'b']) == 123
/Users/shoyer/miniconda/envs/numpy-dev-py3/bin/ipython:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  #!/bin/bash /Users/shoyer/miniconda/envs/numpy-dev-py3/bin/python.app
Out[4]: False
```

There's a good chance @njsmith is already aware of this, but I thought I would note it for future reference. In any case, this will probably be easier to cleanup once we remove the deprecated behavior.

As reported by @amueller on Gitter.

**Comment by njsmith** [comment]
Indeed, see the commit message on: https://github.com/charris/numpy/commit/6bf0e419dc79ea6815557c57b7e9bb504ba20543
and also the giant comment that starts at https://github.com/numpy/numpy/blob/master/numpy/core/src/umath/ufunc_object.c#L871

**Comment by tyrion** [comment]
Is there any update on this? I am still seeing the same behaviour with 1.15.4.

Is there a way to force `numpy.equals` or `==` to perform an elementwise comparison? It would be great to have a workaround while waiting for the "Future" mentioned in the warning to approach :smile:

**Comment by seberg** [comment]
Seems this was at least partially fixed by: 6c25ca1ddfd8d02375544ba9be03b654a62997ba, it now raises an error for the first (`np.equal`) version. Not quite sure that all things around it are fixed, but closing since the initial problem seems gone.
